enhancements/subscription-content/subscription-content-access.md

```diff
@@ -0,0 +1,242 @@
+on the cluster. Each `redhat.repo` file will be saved in a ConfigMap within the `openshift-config`
+namespace, named `redhat-repo-rhel${n}`, where `${n}` is the major version of RHEL associated with
+the `redhat.repo` file.
+
+The cluster subscription manager will also need to obtain the entitlement keys associated with the
+attached OpenShift subscription. It is assumed that OpenShift entitlements grant access to RHEL
+content, but may not grant access to layered content such as EAP or Application Runtimes. This
+content must be saved in a Secret within the `openshift-config` namespace, named
+`etc-pki-entitlement`.
+
+The full details of this component will be specified in a subsequent enhancement proposal.
+
+#### Projected-resource CSI Driver
+
+Once the entitlement key(s) and yum repository definitions are stored on the cluster, the next step
+is to share these definitions across namespaces. This task will fall to the projected-resource
+containers storage interface (CSI) driver. This will be a special storage driver that allows
+ConfigMaps and Secrets to be shared across namespaces via standard pod volume mounts. Access to
+shared Secrets and ConfigMaps will be controlled via namespace-scoped Custom Resources. The
+projected-resource CSI driver will be responsible for conducting a subject access review (SAR) check
+to ensure that the service account for the pod has permission to mount any projected resources.
+
+The full details of this component are specified in the
+[CSI Driver Host Injections proposal](/enhancements/cluster-scope-secret-volumes/csi-driver-host-injections.md)
+
+#### Subscription Content Webhook
+
+The projected-resource CSI driver and associated custom resource are not sufficient to make it
+simple to access subscription content in containers. The task of streamlining access to this content
+will fall to the Subscription Content Webhook. This mutating admission webhook and associated
+operator will do the following:
+
+Define an annotation that can be added to any pod to consume subscription content. The value for
+this annotation will determine which RHEL content will be available. Ex: annotating
+`subscription.openshift.io/version = rhel7` will make RHEL7 content available to the pod’s
+containers. For an annotated pod, the webhook will add the volume and volume mounts to all
+containers such that RHEL content can be consumed The operator for this webhook will create the
+projected-resource CSI driver objects that point to the entitlement keys and yum repo definitions
+for the cluster. The operator will also create RBAC roles that grant permission to the entitlement
+keys and yum repo definitions across namespaces. The keys and repo definitions are assumed to exist
+in the `openshift-config` namespace. These roles will be aggregated to the standard “edit” and
+“admin” roles for the cluster.
+
+The full details of this component will be specified in a future enhancement proposal.
```

Comment:
I wonder if it would help an uninitiated reader if we clarified here that the annotation helps with the higher level abstractions of a pod (Deployment, Build, etc.) that do not expose the volume array of a Pod 

i.e. someone who created a Pod could just do something like 

```
apiVersion: v1
kind: Pod
metadata:
 name: some-pod
spec:
 containers:
   ...
    volumeMounts:
       - name: shared-item-name
         mountPath: /path/to/shared-item-data
 volumes:
     - name: entitlement
       csi:
         driver: projectedresources.storage.openshift.io
         volumeAttributes:
            projectedResourceName: the-projected-resource
```

Reply:
Will clarify this (along with additional refinements I made in drafting #389)